_examples/golang-sse/webapp/src/rpc.gen.ts

```diff
@@ -67,24 +67,32 @@ export class Chat implements Chat {
   
   sendMessage = (args: SendMessageArgs, headers?: object, signal?: AbortSignal): Promise<SendMessageReturn> => {
     return this.fetch(
-      this.url('SendMessage'), createHTTPRequest(args, headers, signal)).then((res) => {
-        return buildResponse(res).then(_data => {
-          return {}
-          })}, (error) => {throw WebrpcRequestFailedError.new({ cause: `fetch(): ${error.message || ''}` })}
-      )
+      this.url('SendMessage'),
+      createHTTPRequest(args, headers, signal)).then((res) => {
+      return buildResponse(res).then(_data => {
+        return {}
+      })
+    }, (error) => {
+      throw WebrpcRequestFailedError.new({ cause: `fetch(): ${error.message || ''}` })
+    })
   }
   
-  
-  subscribeMessages = (args: SubscribeMessagesArgs, options: WebrpcStreamOptions<SubscribeMessagesReturn>): Promise<void> => {
-    const _fetch = () => this.fetch(this.url('SubscribeMessages'),createHTTPRequest(args, options.headers, options.signal)
+  subscribeMessages = (args: SubscribeMessagesArgs, options: WebrpcStreamOptions<SubscribeMessagesReturn>): WebrpcStreamController => {
+    const abortController = options.signal ?? new AbortController();
+
+    const _fetch = () => this.fetch(this.url('SubscribeMessages'),createHTTPRequest(args, options.headers, abortController.signal)
       ).then(async (res) => {
         await sseResponse(res, options, _fetch);
     }, (error) => {
       options.onError(error, _fetch);
     });
-    return _fetch();
+
+    const resp = _fetch();
+    return {
+      abort: abortController.abort.bind(abortController),
+      wait: resp
```

Comment:
since we have `onClose()`, perhaps we don't need this `wait` return argument

or is there a use-case for awaiting stream right away and blocking on the operation? is it idiomatic?

`await stream.closed`

this would have to raise exception, that you can normally get through `onError()`

@AlexanderKolberg thoughts?